cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java

```diff
@@ -84,13 +86,15 @@ public AbstractProgramWorkflowRunner(Arguments runtimeArguments, RunId runId, Pr
    * @return a {@link Callable} of {@link RuntimeContext} for this {@link Program}
    */
   protected Callable<RuntimeContext> getRuntimeContextCallable(String name, final Program program) {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    for (Map.Entry<String, String> systemArgument : systemArguments) {
+      builder.put(systemArgument);
+    }
+    builder.put(ProgramOptionConstants.RUN_ID, runId.getId());
+    builder.put(ProgramOptionConstants.WORKFLOW_BATCH, name);
```

Comment:
This is the issue where we were dropping the existing ``systemArguments`` and recreating then fresh - (lines 89-93 below). Now, workflows accept ``systemArguments`` from a higher layer (example: schedule) and only append more arguments to it, as opposed to recreating the entire ``systemArguments`` afresh. ``LOGICAL_START_TIME`` is already present in the systemArguments passed to this class from a schedule.